### PR TITLE
Fix file not found error when exporting schema with lazy types from CLI

### LINKED ISSUE #2469: Strawberry cli commands fail with error: strawberry.exe\__main__.py not found
After upgrading strawberry to latest version (0.154.1), I am unable to run any strawberry cli commands.

## Describe the Bug
- Upgraded strawberry from 0.152.0 to 0.154.1
```
poetry add strawberry-graphql[debug-server]@0.154.1
```
- Executed below commands:
```
strawberry server myapp.schema
strawberry export-schema myapp.schema:schema
```
- Both these commands are failing in below error:

**FileNotFoundError: [WinError 3] The system cannot find the path specified: 'C:\\Users\\myyuser\\AppData\\Local\\pypoetry\\Cache\\virtualenvs\\straw-k47ybk7v-py3.10\\Scripts\\strawberry.exe\\\_\_main\_\_.py'**

## System Information

 - Operating system: Windows 10
 - Strawberry version (if applicable): 0.154.1
 - Python: 3.10.9

## Additional Context

There is no issue with strawberry cli in version 0.152.0 which I am using currently. If we downgrade the package to this version, cli commands work just fine.

### PR DESCRIPTION
**Pull request by skilkis**
## Types of Changes

- [ ] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Closes #2469 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the CONTRIBUTING document.
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: strawberry/lazy_type.py [hunk 1]
```diff
@@ -48,12 +48,14 @@ def resolve_type(self) -> Type:
...
-                    and module.__file__
-                    and Path(main_module.__file__).samefile(module.__file__)
-                ):
-                    module = main_module
+                main_file = main_module.__file__
+                module_file = module.__file__
+                if main_file and module_file:
+                    try:
+                        is_samefile = Path(main_file).samefile(module_file)
+                    except Exception:
```
Comment (patrick91) [inline strawberry/lazy_type.py:56]: can we be more explicit with the exception? :D 

also I think it might be worth adding a comment on why we do this :)

**Review by patrick91 — COMMENTED** [review]
Thanks for doing this! I've added a minor comment! 😊

File: strawberry/lazy_type.py (hunk 1 above)
Comment (skilkis) [reply strawberry/lazy_type.py:56]: Sure thing! Thanks for already reviewing it

**Comment by botberry** [comment]
Thanks for adding the `RELEASE.md` file!



Here's a preview of the changelog:

---

Fix file not found error when exporting schema with lazy types from CLI #2469


---
Here's the preview release card for twitter:




Here's the tweet text: 
```
🆕 Release (next) is out! Thanks to San Kilkis for the PR 👏

Get it here 👉 https://github.com/strawberry-graphql/strawberry/releases/tag/(next)
```

**Review by patrick91 — APPROVED** [review]

**Pull request merged**